internal/command/output.go

```diff
@@ -5,11 +5,118 @@ package command
 
 import (
 	"bytes"
+	"encoding/json"
+	"fmt"
+	"reflect"
 
 	"github.com/hashicorp/jsonapi"
 )
 
-func outputJson(data interface{}) (string, error) {
+type outputMessage struct {
+	// unique name
+	name string
+	// raw value to return to stdout, platform
+	value interface{}
+	// determines if message should be included in stdout. default: true
+	stdOut bool
+	// determines if message should be sent to platform. default: true
+	platformOut bool
+	// if the value may contain strings/json that is multiline
+	multiLine bool
+}
+
+func (o *outputMessage) IncludeWithPlatform() bool {
+	return o.platformOut
+}
+
+func (o *outputMessage) String() (sValue string) {
```

Comment:
nit: try to avoid named parameters unless you need them for something like `defer foo(&sValue)`.

Also avoid naked returns when using named return parameters, it's less readable than explicitly returning things.

```suggestion
func (o *outputMessage) String() string {
    var sValue string
```